src/search/documents/paper.py

```diff
@@ -134,16 +139,38 @@ def prepare_suggestion_phrases(self, instance):
             # of values that could result in bad suggestions
             weight = int(math.log(instance.unified_document.hot_score_v2, 10) * 10)
 
+        deduped = list(set(phrases))
+        strings_only = [phrase for phrase in deduped if isinstance(phrase, str)]
+
         return {
-            "input": list(set(phrases)),  # Dedupe using set
+            "input": strings_only,  # Dedupe using set
             "weight": weight,
         }
 
+    def prepare_completeness_status(self, instance):
+        try:
+            return instance.get_paper_completeness()
+        except Exception:
+            return Paper.PARTIAL
+
     def prepare_paper_publish_year(self, instance):
         if instance.paper_publish_date:
             return instance.paper_publish_date.year
         return None
 
+    def prepare_can_display_pdf_license(self, instance):
+        try:
+            return pdf_copyright_allows_display(instance)
+        except:
```

Comment:
Not: Avoid bare `except`s:

```suggestion
        except Exception:
```